pythonforandroid/bootstraps/common/build/build.py

```diff
@@ -355,50 +418,78 @@ def make_package(args):
                                   key=LooseVersion)
     build_tools_version = build_tools_versions[-1]
 
+    # Folder name for launcher (used by SDL2 bootstrap)
+    url_scheme = 'kivy'
+
+    # Render out android manifest:
+    manifest_path = "src/main/AndroidManifest.xml"
+    if get_bootstrap_name() != "sdl2":
+        manifest_path = "AndroidManifest.xml"
+    render_args = {
+        "args": args,
+        "service": service,
+        "service_names": service_names,
+        "android_api": android_api
+    }
+    if get_bootstrap_name() == "sdl2":
+        render_args["url_scheme"] = url_scheme
     render(
         'AndroidManifest.tmpl.xml',
-        'src/main/AndroidManifest.xml',
-        args=args,
-        service=service,
-        service_names=service_names,
-        android_api=android_api,
-        url_scheme=url_scheme)
+        manifest_path,
+        **render_args)
 
     # Copy the AndroidManifest.xml to the dist root dir so that ant
     # can also use it
-    if exists('AndroidManifest.xml'):
-        remove('AndroidManifest.xml')
-    shutil.copy(join('src', 'main', 'AndroidManifest.xml'),
-                'AndroidManifest.xml')
-
-    render(
-        'strings.tmpl.xml',
-        'src/main/res/values/strings.xml',
-        args=args,
-        url_scheme=url_scheme,
-        private_version=str(time.time()))
+    if get_bootstrap_name() == "sdl2":
+        if exists('AndroidManifest.xml'):
+            remove('AndroidManifest.xml')
+        shutil.copy(manifest_path, 'AndroidManifest.xml')
 
     # gradle build templates
-    render(
-        'build.tmpl.gradle',
-        'build.gradle',
-        args=args,
-        aars=aars,
-        jars=jars,
-        android_api=android_api,
-        build_tools_version=build_tools_version)
+    if get_bootstrap_name() != "webview":
+        # HISTORICALLY NOT SUPPORTED FOR WEBVIEW. Needs review? -JonasT
+        render(
+            'build.tmpl.gradle',
+            'build.gradle',
+            args=args,
+            aars=aars,
+            jars=jars,
+            android_api=android_api,
+            build_tools_version=build_tools_version)
 
     # ant build templates
-    render(
-        'build.tmpl.xml',
-        'build.xml',
-        args=args,
-        versioned_name=versioned_name)
+    if get_bootstrap_name() != "service_only":
+        # Historically, service_only doesn't support ant anymore.
+        # Maybe we should also drop this for the others? -JonasT
+        render(
+            'build.tmpl.xml',
+            'build.xml',
+            args=args,
+            versioned_name=versioned_name)
+
+    # String resources:
+    if get_bootstrap_name() != "service_only":
```

Comment:
I'm confused why this block isn't shared with the previous one since we have two adjacent `if get_bootstrap_name() != "service_only":`

Reply:
Really just a visual thing to keep the blocks separate. It's kind of random/an accident that both happen to concern only the other bootstraps - since these differences are again historical for not any obviously good reason, and should probably be removed anyway. I could move things around some more to try to get everything concerning one bootstrap closer together, but honestly, I think the whole thing should be revamped fully into some inheritance-style variant anyway. So I can move it around if you want, but I'd rather revamp the whole thing later

Reply:
(The order of things in general is also quite historical, I just tried to minimize changes in overall since the diff/changeset is already massive enough)